src/main.rs

```diff
@@ -562,6 +562,15 @@ fn test_spawn_record_children_subprocesses() {
         }
     }
 
+    // It is possible that the spawned children can become defunct, which may
```

Comment:
This is puzzling because all of the children should be waited for in ruby_forks.rb. How frequently have you seen this after you moved the `wait` call above the results loop?